lib/OpenQA/WebAPI/Controller/API/V1/Search.pm

```diff
@@ -118,6 +118,25 @@ sub _search_perl_modules {
     return \@results;
 }
 
+sub _search_job_modules {
+    my ($self, $keywords, $limit) = @_;
+
+    my @results;
+    my $last_job    = -1;
+    my $like        = {like => "%${keywords}%"};
+    my $job_modules = $self->schema->resultset('JobModules')->search({-or => {script => $like}})->slice(0, $limit);
+    while (my $job_module = $job_modules->next) {
+        my $contents = $job_module->script;
+        if ($job_module->job->id == $last_job) {
```

Comment:
I would avoid repeated calls like `$job_module->job->id` in a potentially big loop like this.